docs/client.rst

```diff
@@ -72,18 +73,118 @@ WBEM Operation recording
+WBEMConnection with the method
+:meth:`~pywbem.WBEMConnection.add_operation_recorder`.
+
+Typical usage scenarios for various operation recorders are the tracing of WBEM
+operations, tracing of interactions with a WBEM server, or the generation of
+test cases.
+
+Please not that the method of activating operation recorders changed starting
+with pywbem 0.11.0 and the addition of a second recorder.  See
+:meth:`~pywbem.WBEMConnection.add_operation_recorder` for more information.
+
+This adds the recorder defined in the method call to a list of active recorders
+in WBEMConnection. All active recorders are called for each WBEMConnection
+method that sends information to the WBEM server.
+
+A single recorder can be  be disabled with
+:meth:`~pywbem.BaseOperationRecorder.disable` method and enabled with
+:meth:`~pywbem.BaseOperationRecorder.enable` method.
+
+The logger names for the operations and http loggers must be
+defined using  the :class:`~pywbem:`PywbemLoggers` class to
+define usable loggers because the pywbem loggers include logging attributes in
+addition to the standard python logger attributes.  This is done with the
+method :meth:`~pywbem.PywbemLoggers.create_logger` which defines one or more
+loggers from the input parameters.  There is also a convience method
+:meth:`~pywbem.PywbemLoggers.create_loggers` which will define multiple pywbem
+logger definitions from a string definition that could be used with command
+line tools.
+
+The following example shows defining loggers for both http and operations and
+adding activating the LogOperationRecorder in WBEMConnection.
+
+::
+
+    # set the python logger facility to output at the DEBUG level
+    logger.setLevel(logging.DEBUG)
+
+    # Define the parameters of the LogOperationRecorder
+    # using one of the methods in the PywbemLoggers class
+    PywbemLoggers.create_logger('all',
+                                log_dest='file',
+                                log_detail_level='min'
+                                log_filename='test.log')   # define both loggers
+
+    # Create the connection and enable the logger
+    conn = WBEMConnection(...)
+    logging = LogOperationRecorder()
+    add_operations_recorder(LogOperationRecorder())
+    # The LogOperationRecorder is now active and writing logs to stderr.
+    # To enable writing logs
+
+The following example activates and enables both recorders:
+
+::
+
+    # To add both recorders
+    logger.setLevel(logging.DEBUG)
+    PywbemLoggers.create_logger('log')   # define only the log logger
+    conn = WBEMConnection(...)
+    add_operations_recorder(LogOperationRecorder())
+    yamlfp = TestClientRecorder.open_file(self.yamlfile, 'a')
+    add_operations_recorder(TestClientRecorder(yamlfp))
+
+    # Both TestClientRecorder and LogOperationRecorder are be
+    # active, enabled and recording
+    # To change the enabled state of either, use the enabled/disabled
+    # methods of the Recorder
+
```

Comment:
The example code above has also some of the issues described for the previous example code.

Reply:
Done